### DISCUSSION TITLE
A 'Migration' is not a registered object？

### DISCUSSION THREAD

**Discussion by biaogxb** [idaholab/moose#20436] [Q&A General] [answered]
when I run the input.i I receive the error：A 'Migration' is not a registered object
If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.

anyone can help me
[Code.zip](https://github.com/idaholab/moose/files/8162435/Code.zip)
？

**Comment by GiudGiud** [comment, marked as answer]
Hello

You forgot to register these objects. See this source file for example, right under the header:
```
registerMooseObject("MooseApp", Diffusion);
```
https://github.com/idaholab/moose/blob/next/framework/src/kernels/Diffusion.C

You ll want to register it to your application

Guillaume

**Reply by biaogxb** [reply]
thank you so much!!!
